pkg/netpol/diff/diff.go

```diff
@@ -119,19 +122,22 @@ func (da *DiffAnalyzer) ConnDiffFromDirPaths(dirPath1, dirPath2 string) (Connect
 	var workloadsNames1, workloadsNames2 map[string]bool
 	var err error
 	if conns1, workloads1, err = caAnalyzer.ConnlistFromDirPath(dirPath1); err != nil {
-		da.errors = append(da.errors, newConnectionsAnalyzingError(err, true, false))
+		// append all fatal/severe errors and warnings returned by caAnalyzer then return because of the fatal err
+		da.appendConnlistErrorsToDiffErrors(caAnalyzer.Errors())
 		return nil, err
 	}
-	da.appendConnlistErrorsToDiffErrors(caAnalyzer.Errors())
-	if da.stopProcessing() {
+	if da.stopProcessing(caAnalyzer.Errors()) {
```

Comment:
why isn't `da.errors` sent to this function?

Reply:
- `stopProcessing()` is called only after returning from a Connlist Analyzer func , so actually we need to check if the connlist Errors list include severe errors and if the `da.stopOnError` flag is on .
- only Connlist Errors may include severe error/ warning beside to fatal (all of the errors raised in diff package / by the Diff analyzer are fatal , and when raised cause to stop running regardless the `stopOnError `flag
- we need to call `stopProcessing() ` after each `caAnalyzer.ConnlistFromDirPath()`, so if there is a severe error and the `stopOnError` flag is on, we need to stop, but we can not append the connlist errors to the diff analyzer errors before calling it, otherwise all connlist errors that were returned from the first `caAnalyzer.ConnlistFromDirPath(dirPath1)` call , will be appended again when returning from the `caAnalyzer.ConnlistFromDirPath(dirPath2)` (if the stopOnError flag was off / if there were only warnings -  that what caused the bug initially)